addons/binding/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/discovery/PulseaudioDeviceDiscoveryService.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.pulseaudio.internal.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.pulseaudio.PulseaudioBindingConstants;
+import org.openhab.binding.pulseaudio.handler.DeviceStatusListener;
+import org.openhab.binding.pulseaudio.handler.PulseaudioBridgeHandler;
+import org.openhab.binding.pulseaudio.handler.PulseaudioHandler;
+import org.openhab.binding.pulseaudio.internal.items.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PulseaudioDeviceDiscoveryService} class is used to discover MAX! Cube
+ * devices that are connected to the Lan gateway.
+ * 
+ * @author Marcel Verpaalen - Initial contribution
+ */
+public class PulseaudioDeviceDiscoveryService extends AbstractDiscoveryService implements DeviceStatusListener {
+
+	private final static Logger logger = LoggerFactory.getLogger(PulseaudioDeviceDiscoveryService.class);
+
+	private PulseaudioBridgeHandler pulseaudioBridgeHandler;
+
+	public PulseaudioDeviceDiscoveryService() {
+		super(PulseaudioHandler.SUPPORTED_THING_TYPES_UIDS, 10, true);
+	}
+
+	public void activate(PulseaudioBridgeHandler pulseaudioBridgeHandler) {
```

Comment:
The pulseaudioBridgeHandler is probably better passed in the constructor directly (to avoid multiple calls of activate() with different instances).